src/Microsoft.Extensions.Testing.Abstractions/StreamingTestDiscoverySink.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.IO;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Extensions.Testing.Abstractions
+{
+    public class StreamingTestDiscoverySink : ITestDiscoverySink
+    {
+        private readonly LineDelimitedJsonStream _stream;
+
+        public StreamingTestDiscoverySink(Stream stream)
+        {
+            _stream = new LineDelimitedJsonStream(stream);
+        }
+
+        public void SendTest(Test test)
```

Comment:
Let's name these after the message type (`TestFound`) so it's easy to match up the messages with what writes them.